unit_test.py

```diff
@@ -223,13 +224,13 @@ def test_valid_topology(self):
         """Test that a valid topology is created"""
         # Arrange
         expected = {'nodes': {'1234abc': {'node_ip': '192.168.1.2',
-                                          'pods': {'dateformat': {'ip': '192.158.1.3', 'ns': 'dockerimage', 'labels': {'guest': 'frontend'}},
-                                                   'kube-router-xfgr': {'ip': '192.168.1.2', 'ns': 'kube-system', 'labels': {}}},
+                                          'pods': {'dateformat': {'ip': '192.158.1.3', 'primary_iface': '','ns': 'dockerimage', 'labels': {'guest': 'frontend'}, 'other_ifaces': {}, 'annotations': {}},
+                                                   'kube-router-xfgr': {'ip': '192.168.1.2', 'primary_iface': '', 'ns': 'kube-system', 'labels': {}, 'other_ifaces': {}, 'annotations': {}}},
                                           'bgp_peers': {'leaf-204': {'prefix_count': 2}}, 'neighbours': {'esxi4.cam.ciscolabs.com':
                                                                                                          {'switches': {'leaf-204': {'vmxnic1-eth1/1'}}, 'Description': 'VMware version 123'}},
-                                          'labels': {'app': 'redis'}, 'mac': 'MOCKMO1C'}},
-                    'services': {'appx': [{'name': 'example service', 'cluster_ip': '192.168.25.5', 'external_i_ps': ['192.168.5.1'],'load_balancer_ip': '192.168.5.2','ns':'appx',
-                                           'labels': {'app': 'guestbook'}}]}}
+                                          'labels': {'app': 'redis'}, 'node_leaf_sec_iface_conn': [], 'node_pod_sec_iface_conn': [], 'node_leaf_ter_iface_conn': [], 'node_pod_ter_iface_conn': [], 'node_leaf_all_iface_conn': [], 'mac': 'MOCKMO1C'}},

```

Comment:
we have added null values everywhere . Is this correct? Should be some value right

Reply:
same comment wherever we are adding null values

Reply:
We have added these additional parameters in our structure. We had to pass these parameters to the already present testcases as well otherwise the testcases would fail.